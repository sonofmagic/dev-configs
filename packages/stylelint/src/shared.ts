import get from 'get-value'
import set from 'set-value'

export function setVscodeSettingsJson(json: any = {}) {
  set(json, 'css\\.validate', false)
  set(json, 'less\\.validate', false)
  set(json, 'scss\\.validate', false)
  const stylelintValidates = new Set(get(json, 'stylelint\\.validate', { default: [] }) as string[])
  stylelintValidates.add('vue')
  stylelintValidates.add('css')
  stylelintValidates.add('scss')
  set(json, 'stylelint\\.validate', Array.from(stylelintValidates))
  return json
}
