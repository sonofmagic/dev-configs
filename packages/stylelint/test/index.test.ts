import { setVscodeSettingsJson } from '@/shared'

describe('index', () => {
  it('setVscodeSettingsJson case 0', () => {
    expect(setVscodeSettingsJson()).toMatchSnapshot()
  })

  it('setVscodeSettingsJson case 1', () => {
    expect(setVscodeSettingsJson({
      'stylelint.validate': [
        'vue',
        'scss',
        'css',
      ],
    })).toMatchSnapshot()
  })
})
