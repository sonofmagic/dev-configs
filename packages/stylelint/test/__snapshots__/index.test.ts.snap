// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`index > common 1`] = `
{
  "extends": [
    "stylelint-config-standard-scss",
    "stylelint-config-recommended-vue/scss",
    "stylelint-config-recess-order",
  ],
  "overrides": [],
  "rules": {
    "scss/at-rule-no-unknown": [
      true,
      {
        "ignoreAtRules": [
          "tailwind",
          "config",
          "unocss",
        ],
      },
    ],
    "scss/selector-no-redundant-nesting-selector": true,
    "selector-type-no-unknown": [
      true,
      {
        "ignoreTypes": [
          "page",
        ],
      },
    ],
    "unit-no-unknown": [
      true,
      {
        "ignoreUnits": [
          "rpx",
        ],
      },
    ],
  },
}
`;

exports[`index > setVscodeSettingsJson case 0 1`] = `
{
  "css.validate": false,
  "less.validate": false,
  "scss.validate": false,
  "stylelint.validate": [
    "vue",
    "css",
    "scss",
  ],
}
`;

exports[`index > setVscodeSettingsJson case 1 1`] = `
{
  "css.validate": false,
  "less.validate": false,
  "scss.validate": false,
  "stylelint.validate": [
    "vue",
    "scss",
    "css",
  ],
}
`;
